luarules/gadgets/game_team_power_watcher.lua

```diff
@@ -0,0 +1,514 @@
+    end
+end
+
+function gadget:UnitDestroyed(unitID, unitDefID, unitTeam)
+    if unitsWithPower[unitID] then
+        unitsWithPower[unitID] = nil
+    end
+    if UnitDefs[unitDefID].power then
+        if teamPowers[unitTeam] then
+            if teamPowers[unitTeam] <= UnitDefs[unitDefID].power then
+                teamPowers[unitTeam] = 0
+            else
+                teamPowers[unitTeam] = teamPowers[unitTeam] - UnitDefs[unitDefID].power
+            end
+        end
+    end
+end
+
+--handles capture events on units already added to unitsWithPower by UnitFinished
+function gadget:MetaUnitAdded(unitID, unitDefID, unitTeam)
+    if unitsWithPower[unitID] then
+        local oldTeam = unitsWithPower[unitID].team
+
+        unitsWithPower[unitID] = {unitID = unitID, power = UnitDefs[unitDefID].power, team = unitTeam}
+        teamPowers[unitTeam] = (teamPowers[unitTeam] + UnitDefs[unitDefID].power) or UnitDefs[unitDefID].power
+
+        if teamPowers[oldTeam] <= UnitDefs[unitDefID].power then
+            teamPowers[oldTeam] = 0
+        else
+            teamPowers[oldTeam] = teamPowers[unitTeam] - UnitDefs[unitDefID].power
+        end
+
+        if teamPowers[unitTeam] and peakTeamPowers[unitTeam] < teamPowers[unitTeam] then
+            peakTeamPowers[unitTeam] = teamPowers[unitTeam]
+        end
+    end
+end
+
+--Returns the power of the input teamID as a number.
+local function teamPower(teamID)
+    for id, power in pairs(teamPowers) do
+        if id == teamID then
+            return power
+        end
+    end
+    return 0
+end
+
+--Returns the total power of all non scavenger/raptor teams as a number.
+local function totalPlayerTeamsPower()
+    local totalPower = 0
+
+    for teamID, power in pairs(teamPowers) do
+        if teamID ~= neutralTeam and teamID ~= scavengerTeam and teamID ~= raptorTeam then
+            totalPower = totalPower + power
+        end
+    end
+
+    return totalPower
+end
+
+--Returns the highest non scavenger/raptor team power as a table {teamID, power}.
+local function highestPlayerTeamPower()
+    local highestPower = 0
+    local highestTeamID = nil
+
+    for teamID, power in pairs(teamPowers) do
+        if teamID ~= neutralTeam and teamID ~= scavengerTeam and teamID ~= raptorTeam then
+            if power > highestPower then
+                highestPower = power
+                highestTeamID = teamID
+            end
+        end
+    end
+
+    return {teamID = highestTeamID, power = highestPower}
+end
+
+--Returns the average of all non scavenger/raptor teams as a number.
+local function averagePlayerTeamPower()
+    local totalPower = 0
+    local teamCount = 0
+
+    for id, power in pairs(teamPowers) do
+        if id ~= neutralTeam and id ~= scavengerTeam and id ~= raptorTeam then
+            totalPower = totalPower + power
+            teamCount = teamCount + 1
+        end
+    end
+
+    local averagePower = totalPower / teamCount
```

Comment:
```suggestion
    local averagePower = teamCount > 0 and totalPower / teamCount or 0
```
This will get a divide-by-zero error in the edge-case of no player teams (e.g. Scavengers vs Gaia)